app/controllers/account/benefits/enrollment/elect.js

```diff
@@ -116,5 +117,27 @@ export default class AccountBenefitsEnrollmentElectController extends Controller
         });
       }
     }
+    this.coverageChangeOccured = true;
+  }
+
+  @action
+  update () {
+    //check if the model is truly dirty
+    if (this.model.enrollment.hasDirtyAttributes || this.coverageChangeOccured) {
+      if (this.updating) {
+        //if we already have a runloop set up, return;
+        return;
+      }
+      this.updating = true;
+
+      //save record in 10 seconds
+      run.later(() => {
```

Comment:
I think we need to track this timer. Also when we destroy / transition, it needs to be cancelled.
```suggestion
      this.updating = run.later(() => {
```

Reply:
Additionally, make the contents of this function a separate function. We shouldn't have anonymous functions in runloop ops where possible.